src/shims/env.rs

```diff
@@ -150,4 +154,30 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             }
         }
     }
+
+    fn update_environ(&mut self) -> InterpResult<'tcx> {
+        let this = self.eval_context_mut();
+        // Collect all the pointers to each variable in a vector.
+        let mut vars: Vec<Scalar<Tag>> = this.machine.env_vars.map.values().map(|&ptr| ptr.into()).collect();
+        // Add the trailing null pointer.
+        vars.push(Scalar::from_int(0, this.pointer_size()));
+        // Make an array with all these pointers inside Miri.
+        let tcx = this.tcx;
+        let environ_layout =
+            this.layout_of(tcx.mk_array(tcx.mk_imm_ptr(tcx.types.u8), vars.len() as u64))?;
```

Comment:
That doesn't seem right... the array should be `vars.len()` many *pointers*, not `vars.len()` many `u8`.

Reply:
Oh wait it's "ptr to `u8`, I see. Never mind.
I was confused because above you used `usize` for the same purpose.^^

Reply:
Yeah, It seems I woke up with different perspectives of the world between these two. I'll keep one of them only then.